doc/research/paf24/perception/VisionNode_Dataflow.md

```diff
@@ -0,0 +1,27 @@
+# VisionNode System Overview
+
+The **VisionNode** system processes camera images and LIDAR data to detect objects and calculate their distances.
+
+## VisionNode System Diagram
+
+![VisionNode System Diagram](../../../assets/VisionNode_Dataflow.PNG)
+(link to lucid chart: [lucid chart](https://lucid.app/lucidchart/34e9aa95-5fb3-4d83-b53f-6d6a3f4748c2/edit?viewport_loc=5190%2C-3952%2C1690%2C703%2C0_0&invitationId=inv_83e27eed-e730-4607-836b-0e863cd2b511))
+
+## Data Flow
+
+1. **Inputs**
+   - **Camera Input (Camera_pub)**: A camera provides RGB image data, which is fed into the VisionNode system.
+   - **LIDAR Input (Lidar_pub)**: A LIDAR sensor supplies distance data used for distance calculations.
+
+2. **Data Handling and Processing**
+   - **Image Handler**: The core processing unit within VisionNode, handling the input data from the camera and the distance data given by the **Distance Handler**. The Image Handler utilizes two main frameworks depending on the given model:
+     - **PyTorch**: Responsible for image segmentation or generating bounding boxes around detected objects. It creates segmentation masks or bounding boxes based on image data.
+     - **Ultralytics**: Focused on object detection, bounding boxes and distance calculations. This module generates bounding boxes for identified objects and calculates distances based on LIDAR input. It also passes data to the traffic light processing
+
+3. **Submodules and Data Processing**
+   - **Traffic Light Processing**: This submodule within the Image Handler processes detected traffic lights. Using distance and image data from object detection, it identifies the position of traffic lights and publishes its image data.
```

Comment:
Does the module not also identify the state of the traffic light? Otherwise the state could not be published (see line 26).

Reply:
I am not sure to be honest. it publishes traffic_light_y_distance aswell as the image on the traffic_light_image topic.
i thought the state detection would run in the traffic light node

Reply:
As far as I understand the vision_node doesn't publish traffic_light_y_distance but the node traffic_light_node does. And traffic_light_image is a message published on the segmented_traffic_light topic.

I think the vision_node currently publishes 6 topics:
- {side}/segmented_image (so 4 topics)
- object_distance
- segmented_traffic_light (the node traffic_light_node subscribes to this topic)

If I'm right, the "Camera_pub" in your image corresponds to the first 4 topics (the segmented_images), the "Object_distance_pub" stands for the object_distance topic and the "Traffic_light_pub" corresponds to the segmented_traffic_light topic.

Currently I don't fully understand your explanation of what Traffic Light Processing (in the image "Process traffic lights") does. You say that the position and image data is published but in the image "Distance, Image" leaves the submodule.
Also: what does "position" mean in this context? Is it the x-y-z-position on the map or the x-y-position in the image? And what is "image data"? Is it a smaller image only containing the traffic light cut from the bigger picture?

Maybe using the topic names used in the code could make the explanation (and especially the picture) more clear.

You also state that the submodule is within the Image Handler but the image does not reflect that. Maybe you could add that in the image?

To sum up my comment: I think the picture needs to be reviewed and the explanation should be more coherent with the picture.